File: packages/installer-server/src/lib/steps/x402Usx.ts
import { execa } from 'execa';
import type { InstallerStep } from './index.js';

export const x402UsxStep: InstallerStep = {
  id: 'x402-usx-skills',
  label: 'Register x402 + USX skills with OpenClaw',
  shouldRun: ({ submission }) =>
    submission.optionalFeatures.solana.x402Skill || submission.optionalFeatures.solana.usxSkill,
  async run({ submission, bus }) {
    const skills: string[] = [];
    if (submission.optionalFeatures.solana.x402Skill) skills.push('@openclaw/skill-x402');
    if (submission.optionalFeatures.solana.usxSkill) skills.push('@solstice/skill-usx');

    for (const pkg of skills) {
      bus.log(x402UsxStep.id, 'info', `Installing skill ${pkg}...`);
      const sub = execa('openclaw', ['tools', 'install', pkg], { all: true });
      sub.all?.on('data', (chunk: Buffer) => {
        const text = chunk.toString().trimEnd();
        if (text) bus.log(x402UsxStep.id, 'info', text);
      });
      await sub;
    }
  },
};

File: internal/installer/steps/x402_usx.go
package steps

import (
	"context"

	"github.com/SolsticeSauer/solsclaw_beta/internal/installer"
)

type X402USX struct{}

func (X402USX) ID() string    { return "x402-usx-skills" }
func (X402USX) Label() string { return "Register x402 + USX skills with OpenClaw" }

func (X402USX) ShouldRun(sc installer.StepContext) bool {
	return sc.Submission.OptionalFeatures.Solana.X402Skill || sc.Submission.OptionalFeatures.Solana.USXSkill
}

func (s X402USX) Run(ctx context.Context, sc installer.StepContext) error {
	var pkgs []string
	if sc.Submission.OptionalFeatures.Solana.X402Skill {
		pkgs = append(pkgs, "@openclaw/skill-x402")
	}
	if sc.Submission.OptionalFeatures.Solana.USXSkill {
		pkgs = append(pkgs, "@solstice/skill-usx")
	}
	for _, pkg := range pkgs {
		sc.Bus.Log(s.ID(), "info", "Installing skill "+pkg)
		if err := streamCommand(ctx, sc, s.ID(), "openclaw", "tools", "install", pkg); err != nil {
			return err
		}
	}
	return nil
}
